### PR TITLE
Adds description to the test action and skeleton readme

### PR DESCRIPTION
**Pull request by chuckbutler**
This is still WIP - however i felt it would be good to get early feedback on the README for the e2e layer, as its got a large role to play and I want it to be well understood going into it:
- How we use it
- Why its important
- How to modify its default behavior
- How to get results and furthermore how to interpret them

### CODE REVIEW

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      type: integer
+  title: test
+  type: object
+```
+
+As an example, you can run a more limited set of tests for rapid validation of
+a deployed cluster:
+
+```shell
+# CODE GOES HERE
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-e2e/README.md]: This is pending completion of my current e-2-e run before i can validate the syntax of changing the regex.

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+inspecting the schema output of the actions:
+
+```shell
+$ juju actions kubernetes-e2e --format=yaml --schema
+test:
+  description: Run end-to-end validation test suite
+  properties:
+    focus:
+      default: \[Conformance\]
+      description: Regex focus for executing the test
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-e2e/README.md]: It's not entirely clear to me what `focus` does - neither the action description nor the README really cover it. An example or further description would help!

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+Action queued with id: 4ceed33a-d96d-465a-8f31-20d63442e51b
+
+$ juju show-action-output 4ceed33a-d96d-465a-8f31-20d63442e51b
+```
+
+## Known issues
+
+The e2e test suite assumes egress network access, as it will pull container
+images from `gcr.io`. You will need to have this registry unblocked in your
+firewall in order to successfully run e2e test results.
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-e2e/README.md]: [Proxy configuration](https://github.com/juju-solutions/bundle-canonical-kubernetes#proxy-configuration) should work too, I'm guessing. Should we call that out?

**Comment by Cynerva** [comment]
:+1: README looks great overall, well done!

File: cluster/juju/layers/kubernetes-e2e/README.md (hunk 3 above)
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-e2e/README.md]: Ah, good point here. While not a magic bullet it certainly does help those edge cases that have a proxy. +1 to this suggestion

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+changes at the system level.
+
+The primary objectives of the e2e tests are to ensure a consistent and reliable
+behavior of the kubernetes code base, and to catch hard-to-test bugs before
+users do, when unit and integration tests are insufficient.
+
+
+## Usage
+
+To deploy the e e2e suite, its best to deploy the
```
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/README.md]: delete 'e'

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+The e2e test is encapsulated as an action to ensure consistent runs of the
+end to end test. The defaults are sensible for most deployments.
+
+```shell
+juju run-action kubernetes-e2e/0 test
+```
+
+### Tuning the e2e test
+
+The e2e test is configurable. By default it will seek to run the gambit of the
```
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/README.md]: In an effort to introduce the parameters use focus and skip in your description.

s/seek to run the gambit/focus on or skip/

File: cluster/juju/layers/kubernetes-e2e/README.md (hunk 2 above)
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/README.md]: I agree that an English sentence about 'focus' and 'skip' would be useful here

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+As an example, you can run a more limited set of tests for rapid validation of
+a deployed cluster:
+
+```shell
+# CODE GOES HERE
+```
+
+### Evaluating end-to-end results
+
+Its not enough to just simply run the test. Result output is stored in two
```
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/README.md]: It is (avoid contractions when writing technical documents).

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+command, as well as a flat file on disk on the `kubernetes-e2e` unit that
+executed the test.
+
+##### Flat file
+
+```shell
+$ juju run-action kubernetes-e2e/0 test
+Action queued with id: 4ceed33a-d96d-465a-8f31-20d63442e51b
+
+$ juju scp kubernetes-e2e/0:4ceed33a-d96d-465a-8f31-20d63442e51b.log .
```
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/README.md]: This shell block leads me to believe we can run the two commands in immediate succession. How does the user know when the test is completed? At least break up the shell block and put "After the tests have completed you can copy the results from the cloud to your computer.

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+##### Action result output
+
+```shell
+$ juju run-action kubernetes-e2e/0 test
+Action queued with id: 4ceed33a-d96d-465a-8f31-20d63442e51b
+
+$ juju show-action-output 4ceed33a-d96d-465a-8f31-20d63442e51b
```
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/README.md]: Again I would not list these two in the same block because an action can take **forever** to run.

File: cluster/juju/layers/kubernetes-e2e/README.md (hunk 5 above)
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/README.md]: Mention that the test pool is very large and the focus and skip parameters help users run the tests that they are interested in.

File: cluster/juju/layers/kubernetes-e2e/README.md (hunk 7 above)
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-e2e/README.md]: Instead of breaking this up, i left the format and instead inserted a note before reaching this content, and called out in bold the potential 1 hour + wait.

**Comment by chuckbutler** [comment]
@Cynerva @mbruzek - I've implemented the review feedback. In some cases not pricely where the comment was left, but in that general viscinity. I feel that this is ready for re-review when you've got time.

**Comment by Cynerva** [comment]
No concerns left here :+1:

**Comment by mbruzek** [comment]
The latest changes look good to me.

File: cluster/juju/layers/kubernetes-e2e/README.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+      default: \[Conformance\]
+      description: Regex focus for executing the test
+      type: string
+    skip:
+      default: \[Flaky\]
+      description: Regex of tests to skip
+      type: string
+    timeout:
+      default: 30000
+      description: Timeout in nanoseconds
```
Comment (wwwtyro) [inline cluster/juju/layers/kubernetes-e2e/README.md]: That seems like a really short timeout - 0.03 ms. Is that what we want for the default? Also, does it have to be in nanoseconds? Can we do a translation, or is operating in ns the norm here?
